File: clienttier/model/generaldata/QuantityResource.js
/**
 * FILE: QuantityResource.js
 * PERCORSO /Codice/sgad/clienttier/model/generaldata/QuantityResource.js
 * DATA CREAZIONE: 16 Febbraio 2014
 * AUTORE: ProTech
 * EMAIL: protech.unipd@gmail.com
 *
 * Questo file è proprietà del gruppo ProTech, viene rilasciato sotto licenza Apache v2.
 *
 * DIARIO DELLE MODIFICHE:
 * 2014-02-17 - Implementazione del metodo valueOf() - Gatto Francesco
 * 2014-02-16 - Creazione della classe - Gatto Francesco
 */
 
 /**
 * Classe per la gestione della quantità di una particolare risorsa.
 * @constructor
 * @param {int} quantity Il numero di unità della data risorsa.
 * @param {Resource} resource La risorsa.
 */
function QuantityResource(quantity, resource) {
	/**
	 * Il numero di unità della data risorsa.
	 * @type {int}
	 * @private 
	 */
	this.quantity = quantity;
	/**
	 * La risorsa
	 * @type {Resource}
	 * @private
	 */
	this.resource = resource;
}
/**
 * Metodo getter della quantità della risorsa.
 * @returns {int} La quantità della risorsa.
 */
QuantityResource.prototype.getQuantity = function() {
    return this.quantity;
};
/**
 * Metodo getter della risorsa.
 * @returns {Resource} La risorsa.
 */
QuantityResource.prototype.getResource = function() {
    return this.resource;
};

/**
 * Ridefinizione del metodo valueOf di QuantityResource.
 * @override
 * @returns {string} Identificativo del QuantityResource.
 */
QuantityResource.prototype.valueOf = function() {
    return JSON.stringify({quantity : this.getQuantity(), resource : this.getResource()});
};

File: Codice/servertier/src/main/scala/sgad/servertier/dataaccess/data/shareddata/QuantityResource.scala
/**
 * FILE: QuantityResource.scala
 * PERCORSO /Codice/sgad/servertier/src/main/scala/sgad/servertier/dataaccess/data/shareddata
 * DATA CREAZIONE: 18 Febbraio 2014
 * AUTORE: ProTech
 * EMAIL: protech.unipd@gmail.com
 *
 * Questo file è proprietà del gruppo ProTech, viene rilasciato sotto licenza Apache v2.
 *
 * DIARIO DELLE MODIFICHE:
 * 2014-02-18 - Creazione della classe - Segantin Fabio
 */
package sgad.servertier.dataaccess.data.shareddata

/**
 * Classe per la gestione della quantità di una particolare risorsa.
 * @constructor
 * @param resource La [[sgad.servertier.dataaccess.data.shareddata.Resource]] di cui si vuole memorizzare la quantità.
 * @param quantity Il numero di unità della data risorsa.
 */
class QuantityResource(private val resource: Resource, private val quantity: Int) {
	if (resource == null
		|| quantity < 0) {
		throw new IllegalArgumentException
	}

	/**
	 * Metodo getter per l'attributo [[sgad.servertier.dataaccess.data.shareddata.QuantityResource.quantity]].
	 * @return La quantità della risorsa in questione.
	 */
	def getQuantity = quantity

	/**
	 * Metodo getter per l'attributo [[sgad.servertier.dataaccess.data.shareddata.QuantityResource.resource]].
	 * @return La risorsa in questione.
	 */
	def getResource = resource

	/**
	 * Metodo per controllare l'uguaglianza.
	 * @param other Un'altra quantità.
	 * @return True o false  in base all'esito del confronto.
	 */
	override def equals(other: Any): Boolean = {
		if (!other.isInstanceOf[QuantityResource])
			false
		else
			(quantity == other.asInstanceOf[QuantityResource].quantity
				&& resource.equals(other.asInstanceOf[QuantityResource].resource)
				)
	}
}